Jenkinsfile

```diff
@@ -43,458 +49,599 @@ pipeline {
+                                                     "${env.GIT_RAW_BASE_URL}/${env.GIT_PREVIOUS_COMMIT}/docker/android/Dockerfile",
+                                                     "${env.GIT_RAW_BASE_URL}/develop/docker/android/Dockerfile",
+                                                     ['PARALLELISM': params.PARALLELISM, 'PLATFORM': params.ABPlatform, 'BUILD_TYPE': params.ABBuildType])
+                sh "curl -L -o /tmp/${env.GIT_COMMIT}/entrypoint.sh ${env.GIT_RAW_BASE_URL}/${env.GIT_COMMIT}/docker/android/entrypoint.sh"
+                sh "chmod +x /tmp/${env.GIT_COMMIT}/entrypoint.sh"
+                iC.inside("-v /tmp/${env.GIT_COMMIT}/entrypoint.sh:/entrypoint.sh:ro -v /tmp/${env.GIT_COMMIT}/bindings-artifact:/tmp/bindings-artifact") {
+                  bindings.doAndroidBindings(params.ABABIVersion)
+                }
               }
             }
           }
           post {
-            success {
-              script {
-                def artifacts = load ".jenkinsci/artifacts.groovy"
-                def commit = env.GIT_COMMIT
-                if (params.JavaBindings) {
-                  javaBindingsFilePaths = [ '/tmp/${GIT_COMMIT}/bindings-artifact/java-bindings-*.zip' ]
-                  artifacts.uploadArtifacts(javaBindingsFilePaths, '/iroha/bindings/java')
-                }
-                if (params.PythonBindings) {
-                  pythonBindingsFilePaths = [ '/tmp/${GIT_COMMIT}/bindings-artifact/python-bindings-*.zip' ]
-                  artifacts.uploadArtifacts(pythonBindingsFilePaths, '/iroha/bindings/python')
+            always {
+              timeout(time: 600, unit: "SECONDS") {
+                script {
+                  try {
+                    if (currentBuild.currentResult == "SUCCESS") {
+                      def artifacts = load ".jenkinsci/artifacts.groovy"
+                      def commit = env.GIT_COMMIT
+                      if (params.JavaBindings) {
+                        javaBindingsFilePaths = [ '/tmp/${GIT_COMMIT}/bindings-artifact/java-bindings-*.zip' ]
+                        artifacts.uploadArtifacts(javaBindingsFilePaths, '/iroha/bindings/java')
+                      }
+                      if (params.PythonBindings) {
+                        pythonBindingsFilePaths = [ '/tmp/${GIT_COMMIT}/bindings-artifact/python-bindings-*.zip' ]
+                        artifacts.uploadArtifacts(pythonBindingsFilePaths, '/iroha/bindings/python')
+                      }
+                      if (params.AndroidBindings) {
+                        androidBindingsFilePaths = [ '/tmp/${GIT_COMMIT}/bindings-artifact/android-bindings-*.zip' ]
+                        artifacts.uploadArtifacts(androidBindingsFilePaths, '/iroha/bindings/android')
+                      }
+                    }
+                  }
+                  finally {
+                    sh "rm -rf /tmp/${env.GIT_COMMIT}"
+                    // cleanWs()
```

Comment:
Please remove if it is not required.

Reply:
@lebdron it was commented out by an accident